### ISSUE TITLE
Full-text index embedding

### ISSUE THREAD

**Issue by ZupoLlask** [py-pdf/pypdf#2645]
Dear Community,

As I haven't found information on this, can you let me know if `pypdf` supports full-text index embedding?
- https://blog.adobe.com/en/publish/2014/07/29/speed-up-pdf-search-with-an-embedded-index

This is super useful for large PDF documents.

Thank you!

**Comment by stefan6419846** [comment]
I just gave it a quick try and it seems like this will modify the root object to include some additional keys afterwards:

```
12 0 obj
<</Lang(de-DE)/Metadata 14 0 R/OpenAction[1 0 R/XYZ null null 0]/Pages 4 0 R/PieceInfo<</SearchIndex<</Index1File 15 0 R/IndexFile 16 0 R/ModID(ec563d98131369489cc90de8634dbe64)/PDXFile 17 0 R>>>>/Type/Catalog>>
endobj
```

Doing a quick search inside the PDF 2.0 specification, I could not find corresponding docs on the search index related fields.

**Comment by ZupoLlask** [comment]
Yeah, I just found it that information is stored under /PieceInfo, usually used for application specific data.

So, probably this search index may be unsupported by non-Acrobat readers...

I'm still researching the subject though.

**Comment by pubpub-zz** [comment]
This is a discussion not an issue. I convert it
